src/frontends/tensorflow/src/op/var_handle.cpp

```diff
@@ -26,22 +28,42 @@ static std::shared_ptr<ov::Node> read_variable(std::shared_ptr<VariablesIndex> v
                                                const ov::Shape shape,
                                                const ::tensorflow::BundleEntryProto& entry,
                                                const NodeContext& node) {
-    std::vector<T> var_data;
     google::protobuf::int64 size = 1;
     for (uint64_t i = 0; i < shape.size(); ++i) {
         size *= static_cast<google::protobuf::int64>(shape[i]);
     }
-    var_data.resize(size);
     TENSORFLOW_OP_VALIDATION(node,
                              size == static_cast<google::protobuf::int64>(entry.size() / sizeof(T)),
                              "[TensorFlow Frontend] Internal error: Available data size isn't equal to calculated.");
-    auto fs = var_index->get_data_file(entry.shard_id());
-    if (!fs.get()) {
-        TENSORFLOW_OP_VALIDATION(node, var_index, "[TensorFlow Frontend] Internal error: Cannot get shard file.");
+    if (var_index->is_mmap_enabled()) {
+        auto mapped_memory = var_index->get_data_mmap(entry.shard_id());
+        if (!mapped_memory.get()) {
+            TENSORFLOW_OP_VALIDATION(node, var_index, "[TensorFlow Frontend] Internal error: Cannot get shard file.");
+        }
+        TENSORFLOW_OP_VALIDATION(
+            node,
+            static_cast<int64_t>(mapped_memory->size()) >= entry.offset() + entry.size(),
+            "[TensorFlow Frontend] Internal error: Variable entry size is out of bounds of mapped memory size.");
+        OPENVINO_SUPPRESS_DEPRECATED_START
+        return std::make_shared<Constant>(
```

Comment:
please avoid `std` since we have `using namespace std`.

Reply:
I would oppositely avoid using `using namespace std` 😄

Reply:
it uses in `cpp` not `hpp`. It should be safe